coordinator/persistence-cassandra-5.0/src/main/java/io/stargate/db/cassandra/impl/Conversion.java

```diff
@@ -0,0 +1,621 @@
+import org.apache.cassandra.stargate.exceptions.TruncateException;
+import org.apache.cassandra.stargate.exceptions.UnauthorizedException;
+import org.apache.cassandra.stargate.exceptions.UnavailableException;
+import org.apache.cassandra.stargate.exceptions.WriteFailureException;
+import org.apache.cassandra.stargate.exceptions.WriteTimeoutException;
+import org.apache.cassandra.stargate.transport.ProtocolException;
+import org.apache.cassandra.stargate.transport.ProtocolVersion;
+import org.apache.cassandra.stargate.transport.ServerError;
+import org.apache.cassandra.stargate.utils.MD5Digest;
+import org.apache.cassandra.transport.Event;
+import org.apache.cassandra.transport.messages.ResultMessage;
+import org.apache.cassandra.utils.NoSpamLogger;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class Conversion {
+
+  private static final Logger logger = LoggerFactory.getLogger(Conversion.class);
+  private static final NoSpamLogger noSpamLogger =
+      NoSpamLogger.getLogger(logger, 5L, TimeUnit.MINUTES);
+
+  // This LivenessInfo is used only for constructing intermediate `Row` objects so as to build
+  // a `PagingState`, but it does not actually go into the `PagingState` object
+  private static final LivenessInfo DUMMY_LIVENESS_INFO = LivenessInfo.create(0, 0);
+
+  private static final Map<Class<? extends AbstractType>, Column.Type> TYPE_MAPPINGS;
+
+  static {
+    Map<Class<? extends AbstractType>, Column.Type> types = new HashMap<>();
+    types.put(AsciiType.instance.getClass(), Column.Type.Ascii);
+    types.put(LongType.instance.getClass(), Column.Type.Bigint);
+    types.put(BytesType.instance.getClass(), Column.Type.Blob);
+    types.put(ByteType.instance.getClass(), Column.Type.Tinyint);
+    types.put(BooleanType.instance.getClass(), Column.Type.Boolean);
+    types.put(CompositeType.class, Column.Type.Composite);
+    types.put(CounterColumnType.instance.getClass(), Column.Type.Counter);
+    types.put(SimpleDateType.instance.getClass(), Column.Type.Date);
+    types.put(DecimalType.instance.getClass(), Column.Type.Decimal);
+    types.put(DoubleType.instance.getClass(), Column.Type.Double);
+    types.put(DurationType.instance.getClass(), Column.Type.Duration);
+    types.put(DynamicCompositeType.class, Column.Type.DynamicComposite);
+    types.put(FloatType.instance.getClass(), Column.Type.Float);
+    types.put(InetAddressType.instance.getClass(), Column.Type.Inet);
+    types.put(Int32Type.instance.getClass(), Column.Type.Int);
+    types.put(IntegerType.instance.getClass(), Column.Type.Varint);
+    types.put(ShortType.instance.getClass(), Column.Type.Smallint);
+    types.put(UUIDType.instance.getClass(), Column.Type.Uuid);
+    types.put(UTF8Type.instance.getClass(), Column.Type.Text);
+    types.put(TimeType.instance.getClass(), Column.Type.Time);
+    types.put(TimestampType.instance.getClass(), Column.Type.Timestamp);
+    types.put(TimeUUIDType.instance.getClass(), Column.Type.Timeuuid);
```

Comment:
Need to add VectorType

Reply:
I thought first that it should be, but then remembered it is supported at a later point, see next comment in this class.